src/main/java/net/dv8tion/jda/api/interactions/commands/build/OptionData.java

```diff
@@ -724,6 +756,87 @@ public OptionData setRequiredRange(double minValue, double maxValue)
+        if (type != OptionType.STRING)
+            throw new IllegalArgumentException("Can only set min length for options of type STRING");
+        Checks.notNegative(minLength, "Min length");
+        this.minLength = minLength;
+        return this;
+    }
+
+    /**
+     * Configure the maximum length for strings which can be provided for this option.
+     *
+     * @param  maxLength
+     *         The maximum length for strings which can be provided for this option.
+     * @throws IllegalArgumentException
+     *         If any of the following checks fail
+     *         <ul>
+     *             <li>{@link OptionType type of this option} is {@link OptionType#STRING STRING}</li>
+     *             <li>{@code maxLength} is positive and lower than or equal to {@value MAX_STRING_OPTION_LENGTH}</li>
+     *         </ul>
+     *
+     * @return The OptionData instance, for chaining
+     */
+    @Nonnull
+    public OptionData setMaxLength(int maxLength)
+    {
+        if (type != OptionType.STRING)
+            throw new IllegalArgumentException("Can only set max length for options of type STRING");
+        Checks.positive(maxLength, "Max length");
+        Checks.check(maxLength <= MAX_STRING_OPTION_LENGTH, "Max length must not be greater than %d. Provided: %d", MAX_STRING_OPTION_LENGTH, maxLength);
+        this.maxLength = maxLength;
+        return this;
+    }
+
+    /**
+     * Configure the minimum and maximum length for strings which can be provided for this option.
+     *
+     * @param  minLength
+     *         The minimum length for strings which can be provided for this option.
+     * @param  maxLength
+     *         The maximum length for strings which can be provided for this option.
+     * @throws IllegalArgumentException
+     *         If any of the following checks fail
+     *         <ul>
+     *             <li>{@link OptionType type of this option} is {@link OptionType#STRING STRING}</li>
+     *             <li>{@code minLength} is not negative</li>
+     *             <li>{@code maxLength} is lower than or equal to {@value MAX_STRING_OPTION_LENGTH}</li>
+     *             <li>{@code minLength} is lower than or equal to {@code maxLength}</li>
+     *         </ul>
+     *
+     * @return The OptionData instance, for chaining
+     */
+    @Nonnull
+    public OptionData setRequiredLength(int minLength, int maxLength)
+    {
+        if (type != OptionType.STRING)
+            throw new IllegalArgumentException("Can only set min and max length for options of type STRING");
+        Checks.notNegative(minLength, "Min length");
+        Checks.check(maxLength <= MAX_STRING_OPTION_LENGTH,
+                "Max length must not be greater than %d. Provided: %d", MAX_STRING_OPTION_LENGTH, maxLength);
+        Checks.check(minLength <= maxLength, "Min length must not be greater than max length. Provided: %d > %d", minLength, maxLength);
+        this.minLength = minLength;
+        this.maxLength = maxLength;
```

Comment:
Is there a reason we re-implement these checks instead of doing:
```suggestion
        Checks.check(minLength <= maxLength, "Min length must not be greater than max length. Provided: %d > %d", minLength, maxLength);
        this.setMinLength(minLength);
        this.setMaxLength(maxLength);
```